core/src/main/java/io/substrait/dsl/SubstraitBuilder.java

```diff
@@ -426,6 +459,12 @@ public Aggregate.Grouping grouping(Rel input, int... indexes) {
     return Aggregate.Grouping.builder().addAllExpressions(columns).build();
   }
 
+  public Aggregate.Grouping grouping(Expression... expressions) {
+    List<Expression> exprs =
+        Arrays.stream(expressions).collect(java.util.stream.Collectors.toList());
+    return Aggregate.Grouping.builder().addAllExpressions(exprs).build();
```

Comment:
I don't think this works because `addAllExpressions` requires an `Iterable<? extends Expression> elements`, not an   `Iterator<Expression>`.

That being said, there is an `addExpressions(Expression... elements)` methods that can be used directly 😅 .

Reply:
fwiw `() -> Arrays.stream(expressions).iterator()` _is_ an `Iterable`, since it is a lambda which, when called, produces an `Iterator`.